src/core/bsearch.rkt

```diff
@@ -94,15 +94,13 @@
 
   (define eval-expr
     (eval-prog `(λ ,(program-variables (alt-program (car alts))) ,expr) 'fl ctx))
-
   (define var (gensym 'branch))
   (define ctx* (context-extend ctx var repr))
   (define progs (map (compose (curryr extract-subexpression var expr) alt-program) alts))
   (define start-prog (extract-subexpression (*start-prog*) var expr))
-
   ; Not totally clear if this should actually use the precondition
   (define precondition `(λ ,(program-variables start-prog) (TRUE)))
-  (define start-fn (make-search-func precondition (list start-prog) ctx*))
```

Comment:
Is it better to use (list ctx*) or (cons ctx* `()) 
This is used in a few places

Reply:
We try to use `(cons foo '())` if we are thinking about it as a pair (pair of something and an empty list) and otherwise we'd do `(list foo)`. Basically using coding convention to make up for the lack of reasonable types.